src/compression_static.rs

```diff
@@ -30,7 +33,7 @@ pub async fn precompressed_variant(
 
     // Try to find the pre-compressed metadata variant for the given file path
     if let Some(ext) = precomp_ext {
-        let mut filepath_precomp = file_path;
+        let mut filepath_precomp = file_path.to_owned();
         if let Some(filename) = filepath_precomp.file_name() {
             let filename = filename.to_str().unwrap();
             let precomp_file_name = [filename, ".", ext].concat();
```

Comment:
Nice!
Just looking into the code involved I realized a couple of minor improvements that we can possibly do too.

- In line 38, handling a potential panic when doing the `&str` convention.
- In line 39, reducing one allocation when extending the same UTF-8 `&str` via `String::from`. Also because `contact` is ambiguous.

I can not do the "next lines changes" suggestion directly in the GitHub review tool but feel free to incorporate/improve this below.

```rs
            if let Some(filename) = filename.to_str() {
                let mut precomp_filename = String::from(filename);
                precomp_filename.push_str(".");
                precomp_filename.push_str(ext);
                filepath_precomp.set_file_name(precomp_filename);

                tracing::trace!(
                    "getting metadata for pre-compressed file variant {}",
                    filepath_precomp.display()
                );

                if let Ok((meta, _)) = file_metadata(&filepath_precomp) {
                    tracing::trace!("pre-compressed file variant found, serving it directly");

                    let encoding = if ext == "gz" { "gzip" } else { ext };
                    precompressed = Some((filepath_precomp, meta, encoding));
                }
            }
```

Reply:
Don't worry I just fired a commit 😅